src/arrayMethodSort.js

```diff
@@ -1,11 +1,27 @@
 'use strict';
 
-/**
- * Implement method Sort
- */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const ARRAY = [...this];
+    let cmpFunc = compareFunction;
+
+    if (!compareFunction) {
+      cmpFunc = (a, b) => String(a) > String(b);
+    }
+
+    for (let i = 0; i < ARRAY.length - 1; i++) {
+      for (let j = 0; j < ARRAY.length - 1 - i; j++) {
+        if (cmpFunc(ARRAY[j], ARRAY[j + 1]) > 0) {
+          [ARRAY[j], ARRAY[j + 1]] = [ARRAY[j + 1], ARRAY[j]];
+        }
+      }
+    }
```

Comment:
The current implementation uses a bubble sort algorithm, which has O(n^2) time complexity and is not efficient for large arrays. While it is acceptable for this task, you might consider exploring more efficient sorting algorithms like quicksort or mergesort for better performance with larger data sets.